### PR TITLE
IMP sceKernelFindModuleByUID and sceKernelFindModuleByName.

### PR DESCRIPTION
**Pull request by shenweip**
This should help some patched games like #13205, #13204. I think @sum2012 would be interested in this.

### CODE REVIEW

File: Core/HLE/sceKernelModule.cpp [hunk 1]
```diff
@@ -263,11 +269,23 @@ class PSPModule : public KernelObject {
...
+			NativeModule *pnm = &nm;
+			char *ptemp = temp;
+			DoArray(p, ptemp, 0xC0);
+			memcpy(pnm, ptemp, 0x2C);
+			pnm->modid = GetUID();
+			pnm += 0x30;
+			ptemp += 0x30;
+			memcpy(pnm, ptemp, 0xC0 - 0x2C);
+		}
 		Do(p, memoryBlockAddr);
```
Comment (aliaspider) [inline Core/HLE/sceKernelModule.cpp:289]: why not just put 'modid' last in 'NativeModule' to keep the struct backwards compatible and avoid this hack ? or is 'NativeModule'  something the real PSP OS uses and it just had wrong offsets before ?

Comment (shenweip) [reply Core/HLE/sceKernelModule.cpp:289]: NativeModule is a native struct of PSP, its members has fixed order.

**Comment by sum2012** [comment]
Well done @shenweip 
jpcsp use 3 tables to make conversion
https://github.com/jpcsp/jpcsp/blob/master/src/jpcsp/HLE/kernel/managers/ModuleManager.java
But we have another method

File: Core/HLE/sceKernelModule.cpp (hunk 1 above)
Comment (aliaspider) [reply Core/HLE/sceKernelModule.cpp:289]: ok I see.
why are you increasing both `pnm`and `ptemp` by `0x30` though ? shouldn't `ptemp`be increased by `0x2C` instead ?

Comment (shenweip) [reply Core/HLE/sceKernelModule.cpp:289]: Oops, thanks.

**Comment by sum2012** [comment]
Well done @shenweip Fixed #13205 #13204 

I just want a small change
INFO_LOG(SCEMODULE, "sceKernelFindModuleByUID(%d)", uid); change to
INFO_LOG(SCEMODULE, "%d = sceKernelFindModuleByUID(%d)", module->modulePtr, uid);

File: Core/HLE/sceKernelModule.cpp (hunk 1 above)
Comment (hrydgard) [reply Core/HLE/sceKernelModule.cpp:289]: Turns out the math was wrong - have to cast the pointers to bytes first :)

Fixed in 2f4dfed

Comment (shenweip) [reply Core/HLE/sceKernelModule.cpp:289]: Oops again, thanks:p

**Pull request merged**